### DISCUSSION TITLE
Security concern

### DISCUSSION THREAD

**Discussion by rustdesk** [rustdesk/rustdesk#3982] [Ideas]
Received from email.

```
1) My phone had priority, couldn’t move my mouse when I was controlled

2) No  shortcut to disconnect ? So I pressed Alt f4, it colsed the connection window

3) The phone still had control, so if this was a real situation I was fully stuck

4) I right click on the system tray app and stoped it, the phone STILL HAD CONTROL

5) Launch the app back, it was written than I was disconnect, the phone is still in control
```

**Comment by rustdesk** [comment]
4), yes, should close connections while turning off service.

**Comment by My1** [comment]
I'll add:

6) closing the mini window (e.g. from the task bar) also doesnt seem to disconnect.

regarding 1) the only way to really solve that is using a timeout as the controller's mouse cursor behaves like a finger on a touchscreen and the cusor of the computer that is controlled just teleports over.

like in anydesk, generally when the person in front of the computer moves the mouse a timeout of roughly 1-2 seconds is applied in which the remote person's mouse gets moved to show-mode basically their mouse cursor is shown in a different color and can be used to show stuff but not actually do anything, just like when control isnt allowed in the first place.

so I usually tell ppl on the phone that they either need to let go of the mouse to let me work, or if I do something they feel uncomfortable with, they can intervene at ANY time to take over and disconnect.

**Comment by rustdesk** [comment]
@21pages check out if our new version closes all connections if close cm from taskbar

**Comment by korden32** [comment]
@rustdesk Windows, RustDesk 1.2.0 nightly (commit 52ce3dd) from releases still has same behavior - stopping service and closing connection window by "X" doesn't close existing connection. 

Note: it closes when you click "Close" in taskbar context menu, but not when you left-click on taskbar icon and press "X" in the connection window.

**Comment by loudar** [comment]
Is there any update on the priority from the controlling instance? This seems like an obvious design flaw. If desired, the controlled desktop should always have priority and full control over the mouse. Great entry point for scammers 😬

**Comment by My1** [comment]
Well it isn't as if remote priority is a design feature or anything, it's just by the way that the remote mouse cursor is decoupled from the local mouse cursor.

Basically its the same as if you used a touch screen and a mouse at the same time.

Whenever you touch, the mouse is gonna teleport over to that location and the mouse is pretty much defenseless.

Obviously there needs to be some way to deal with that, like for example a timeout.

**Comment by rustdesk** [comment]
@fufesou checkout https://github.com/rustdesk/rustdesk/blob/master/src/ui/remote.tis#L178 we may need better got_mouse_control, currently only check if moving big distance, we may also add some timeout as above mentioned, when the mouse is moved by the other side, we need a small timeout to get control. please only do on Flutter version.

**Comment by fufesou** [comment]
I've given mouse a protection timeout. Self-side cann't aquire mouse control when peer-side is active.

But for now, peer-side cann't aquire mouse control when self-side is active. I'm trying to fix this problem.

https://github.com/rustdesk/rustdesk/pull/2007

**Comment by rustdesk** [comment]
> @rustdesk Windows, RustDesk 1.2.0 nightly (commit [52ce3dd](https://github.com/rustdesk/rustdesk/commit/52ce3dd2c299e262a54af0b5350fa60f66215e60)) from releases still has same behavior - stopping service and closing connection window by "X" doesn't close existing connection.
> 
> Note: it closes when you click "Close" in taskbar context menu, but not when you left-click on taskbar icon and press "X" in the connection window.

@korden32 could you test again？

**Comment by rustdesk** [comment]
> Well it isn't as if remote priority is a design feature or anything, it's just by the way that the remote mouse cursor is decoupled from the local mouse cursor.
> 
> Basically its the same as if you used a touch screen and a mouse at the same time.
> 
> Whenever you touch, the mouse is gonna teleport over to that location and the mouse is pretty much defenseless.
> 
> Obviously there needs to be some way to deal with that, like for example a timeout.

Could you test? new nightly build will be generated later.

**Comment by rustdesk** [comment]
> Could you test? new nightly build will be generated later.

@My1
